common/metrics/otel_metric_test_utility.go

```diff
@@ -40,24 +46,22 @@ var _ OpentelemetryReporter = (*TestOtelReporter)(nil)
 
 type (
 	TestOtelReporter struct {
-		MeterProvider *metrictest.MeterProvider
```

Comment:
The test implementation of Meter and MeterProvide was also removed in v0.28.0. There's an open issue for adding a test impl and we can switch to that when the linked pr is landed and released.